hydroshare/settings.py

```diff
@@ -566,4 +572,107 @@
 TRACKING_USER_FIELDS = ["username", "email", "first_name", "last_name"]
 
 # info django that a reverse proxy sever (nginx) is handling ssl/https for it
-SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
\ No newline at end of file
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+
+# Content Security Policy
+# See http://django-csp.readthedocs.io/en/latest/configuration.html#configuration-chapter
+# sha256-* strings are hashes of inline scripts and styles
```

Comment:
How are these hashes calculated and recalculated on code changes? Could some of them be moved out to their own JS files?